src/com/pentalog/pentastagiu/homework2/shop/Basket.java

```diff
@@ -0,0 +1,52 @@
+package com.pentalog.pentastagiu.homework2.shop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Basket {
+    private static int totalPrice;
+    private static Customer customerBakset;
+    public static List<Product> basketProductsList= new ArrayList<>();
+
+    public Basket(Customer customer) {
+        this.customerBakset = customer;
+        this.totalPrice = 0;
+    }
+
+    public  int getTotalPrice(){
+        if(customerBakset.getUserMembership() == Customer.Membership.GOLD){
```

Comment:
dar daca un customerBasket e null? trebuie sa adaugi validare pentru asta ori aici sau si mai bine in constructor
ar fi fost protrivit sa folosesti un switch pe enum si sa pastrezi in enum si valoarea discountului(prin folosirea unui field private in enum) si astfel ai avea legatura asta intr-un singur loc si nu te va incurca/nu va fi complicat de extins, nu conteaza in cate locuri vei vrea sa te folosesti de membership/discount